docs/adr/0003-pre-commit-hooks-for-code-quality.md

```
@@ -0,0 +1,153 @@
+The current workflow requires developers to manually run:
+```bash
+ruff format .
+ruff check .
+mypy src/openehr_sdk
+```
+
+This is error-prone and often forgotten, leading to failed CI runs and additional commits just for formatting fixes.
+
+## Decision
+
+We will investigate and implement pre-commit hooks using the `pre-commit` framework (https://pre-commit.com/) to automatically run code quality checks before allowing commits.
```

markdownlint-cli2 (0.18.1)

Bare URL used

(MD034, no-bare-urls)